rsrc/sourcegraph.js

```diff
@@ -0,0 +1,62 @@
+/**
+* @provides sourcegraph
+*/
+
+window.SOURCEGRAPH_PHABRICATOR_EXTENSION = true;
+
+/**
+ * Scrapes a Phabricator username from the DOM.
+ */
+function getPhabricatorUsername() {
+  var USERNAME_URL_PATTERN = /\/p\/([A-Z0-9-_]+)/i;
+  var coreMenuItems = document.getElementsByClassName(
+    'phabricator-core-user-menu'
+  );
+  for (var i = 0; i < coreMenuItems.length; i++) {
+    var coreMenuItem = coreMenuItems.item(i);
+    var possiblePersonUrl = coreMenuItem.getAttribute('href');
+    if (!possiblePersonUrl) {
+      continue;
+    }
+    var match = USERNAME_URL_PATTERN.exec(possiblePersonUrl);
+    if (!match) {
+      continue;
+    }
+    return match[1];
+  }
+  return null;
+}
+
+/**
+ * To prevent loading the extension for all users, specifify a user whitelist
+ * by changing this line to `var userWhitelist = { "username": true, ... };`
+ */
+var userWhitelist = undefined;
```

Comment:
Not sure if we should address this in this PR but this variable was only here to provide a way for admins to manually specify a whitelist of users. Since the goal is to make the installation and management as simple as possible, this should eventually be a 1st class configuration option in the phabricator extension

Reply:
I'll add it to the configuration list. It might make since for this to be an admin feature only. Since the `sourcegraph.url` conduit endpoint is available for all authed users I'll make admin specific ones so that @sqs can add his access_token work there instead.